backend/app/spread/tests/test_update_balances_with_spread.py

```diff
@@ -0,0 +1,126 @@
+    )
+    spread_balances = update_balances_with_spread(
+        balances, "GVT", "USDC", 20, "2022-01-01", "2022-01-02"
+    )
+    assert "GVT" in spread_balances.balances
+    assert "USDC" in spread_balances.balances
+    assert (
+        spread_balances.balances["GVT"].loc["2022-01-01"]
+        + spread_balances.balances["USDC"].loc["2022-01-01"]
+        == 123
+    )
+    assert spread_balances.balances["USDC"].loc["2022-01-01"] == pytest.approx(
+        123 * 20 / 100.0
+    )
+
+
+def test_token_to_usdc_earlier_start():
+    "Test correct behavior when start date is before any first balance value"
+    balances = Balances(
+        balances={"GVT": _make_balance_series([123, 246], ["2022-01-02", "2022-01-03"])}
+    )
+    spread_balances = update_balances_with_spread(
+        balances, "GVT", "USDC", 20, "2022-01-01", "2022-01-03"
+    )
+    assert "GVT" in spread_balances.balances
+    assert "USDC" in spread_balances.balances
+    assert spread_balances.balances["USDC"].loc["2022-01-01"] == 0
+
+
+def test_token_to_existing_usdc():
+    balances = Balances(
+        balances={
+            "GVT": _make_balance_series([123, 246], ["2022-01-01", "2022-01-02"]),
+            "USDC": _make_balance_series([10, 10], ["2022-01-01", "2022-01-02"]),
+        }
+    )
+    spread_balances = update_balances_with_spread(
+        balances, "GVT", "USDC", 20, "2022-01-01", "2022-01-02"
+    )
+    assert "GVT" in spread_balances.balances
+    assert "USDC" in spread_balances.balances
+    assert (
+        spread_balances.balances["GVT"].loc["2022-01-01"]
+        + spread_balances.balances["USDC"].loc["2022-01-01"]
+        == 133
+    )
+    assert spread_balances.balances["USDC"].loc["2022-01-01"] == pytest.approx(
+        10 + 123 * 20 / 100.0
+    )
+
+
+def test_token_to_exising_usdc_different_dates():
+    "When existing USDC balance series begins later than start date"
+    balances = Balances(
+        balances={
+            "GVT": _make_balance_series(
+                [123, 246, 123, 246],
+                [
+                    "2022-01-01",
+                    "2022-01-02",
+                    "2022-01-03",
+                    "2022-01-04",
+                ],
+            ),
+            "USDC": _make_balance_series(
+                [10, 10],
+                [
+                    "2022-01-03",
+                    "2022-01-04",
+                ],
+            ),
+        }
+    )
+    spread_balances = update_balances_with_spread(
+        balances, "GVT", "USDC", 20, "2022-01-01", "2022-01-04"
+    )
+    assert "GVT" in spread_balances.balances
+    assert "USDC" in spread_balances.balances
+    assert (
+        spread_balances.balances["GVT"].loc["2022-01-01"]
+        + spread_balances.balances["USDC"].loc["2022-01-01"]
+        == 123
+    )
+    assert spread_balances.balances["USDC"].loc["2022-01-01"] == pytest.approx(
+        123 * 20 / 100.0
+    )
```

Comment:
Maybe we should also show the spread balances at  "2022-01-03" - "2022-01-04" as well. See previous comment.